### PR TITLE
Set HTTP Method for `_session` requests

### PR DESCRIPTION
**Pull request by rhyshort**
## What
Fix issue where session authentication would fail.

## How

Set the HTTP method for `_session` requests made by the `CDTSessionCookieInterceptor` 

## Testing

The OHHTTPStub now checks for the correct HTTP method and delivers a response based on it. For `GET` requests it will return 200 with an empty body, and no headers, for `POST` the cookie header will be set and a 200 response returned.

## Reviewers

reviewer @tomblench 
reviewer @mikerhodes

### CODE REVIEW

**Comment by tomblench** [comment]
+1

File: Tests/Tests/CDTSessionCookieInterceptorTests.m [hunk 1]
```diff
@@ -49,18 +49,23 @@ - (void)setUp
...
+                      @"roles" : @[ @"_admin" ]
+                  }
+                              statusCode:200
+                                 headers:@{
+                                     @"Set-Cookie" : [NSString
+                                         stringWithFormat:@"%@; Version=1; Path=/; HttpOnly",
+                                                          testCookieHeaderValue]
+                                 }];
+          } else {
+              return [OHHTTPStubsResponse responseWithJSONObject:@{} statusCode:200 headers:@{}];
```
Comment (mikerhodes) [inline Tests/Tests/CDTSessionCookieInterceptorTests.m]: I'd check whether `200` is really what Couch sends here.

Comment (mikerhodes) [inline Tests/Tests/CDTSessionCookieInterceptorTests.m]: Or can you just somehow fail if the exact expected request isn't met -- you probably want to check for the form-encoded body too?

Comment (rhyshort) [inline Tests/Tests/CDTSessionCookieInterceptorTests.m]: Couch returns `200` for a post to session see: http://docs.couchdb.org/docs/couchdb/en/stable/api/server/authn.html#session.

I can't check the body to make sure it is form encoded, so far my attempts to read HTTPBodyStream always fail. The only thing I can check is the content type

Comment (mikerhodes) [inline Tests/Tests/CDTSessionCookieInterceptorTests.m]: This is in the `else` block so it's not a POST to session. It looks like it'll return some info about the user whose details are in the request with a `200` code from the docs.

Comment (rhyshort) [inline Tests/Tests/CDTSessionCookieInterceptorTests.m]: I think I understand what you mean, and I think I implemented it in 24bb3b5

**Comment by mikerhodes** [comment]
+1

**Pull request merged**